tests/suites/storage/persistent_storage.sh

```diff
@@ -0,0 +1,145 @@
+	echo "Check properties of single block device storage unit: PASSED."
+
+	#
+	# check type, persistent setting and pool of single filesystem storage unit
+	#
+	juju storage --format json | jq '.storage | ."single-fs/1" | .kind' | check "filesystem"
+	# persistent setting
+	echo "Checking persistent setting of single filesystem storage unit"
+	juju storage --format json | jq '.storage | ."single-fs/1" | .persistent' | check false
+	# pool storage
+	juju storage --format json | jq '.volumes | ."0" | .storage' | check "single-blk/0"
+	# pool setting
+	juju storage --format json | jq '.volumes | ."0" | .pool' | check "ebs"
+	# storage status
+	juju storage --format json | jq '.volumes | ."0" | .status | .current' | check "attached"
+	echo "Check properties of single filesystem storage unit: PASSED."
+
+	# assert charm removal message for single block and filesystem storage
+	removal_msg=$(juju remove-application dummy-storage 2>&1)
+	echo "${removal_msg}" | sed -sn 2p | sed 's/^-//' | check "will remove storage single-fs/1"
+	echo "${removal_msg}" | sed -sn 3p | sed 's/^-//' | check "will detach storage single-blk/0"
+	#
+	# wait until an update of storage status occurred. Due to the asynchronous nature of Juju,
+	# the status of storage may change time to time after a Juju CLI issued, in this test only
+	# the existence of storage id is the point of interest.
+	#
+	wait_for "{}" ".applications" # we use this wait_for command as an indicator that the storage
+	# status has changed and now we can check for the storage status and assert that indeed the
+	# single filesystem storage unit has been removed successfully.
+	juju storage --format json | jq '.storage | has("single-fs/1")' | check false
+
+	echo "Checking total number of storage unit(s)."
+	juju storage --format json | jq '.storage | keys | length' | check 1
+	echo "Check for existence of single block storage"
+	juju storage --format json | jq '.storage | has("single-blk/0")' | check true
+	echo "single-blk/0 found in storage list."
+
+	echo "Check for existence of single-blk/0 persistent storage after remove-application"
+	juju storage --format json | jq '.volumes | ."0" | .storage' | check "single-blk/0"
+	# storage status
+	juju storage --format json | jq '.volumes | ."0" | .status | .current' | check "detached"
+	echo "Check status of persistent storage single-blk/0 after remove-application: PASSED"
+
+	# Deploy charm with an existing detached storage
+	juju deploy -m "${model_name}" ./testcharms/charms/dummy-storage --attach-storage single-blk/0
+	# wait for current application-status to be active
+	wait_for "active" '.applications | .[] | ."application-status" | .current'
+	# wait for current workload-status to be active
+	wait_for "active" "$(workload_status "dummy-storage" 1).current"
+	echo "Checking current status of app dummy-storage."
+	juju status --format json | jq -S '.applications | .[] | ."application-status" | .current' | check "active"
```

Comment:
Similar here. I think this is all just the same as `wait_for "dummy-storage" $(idle_condition "dummy-storage")`